### ISSUE TITLE
OpenApi: Enum as implementation class inside @Schema

### ISSUE THREAD

**Issue by srnjak** [OpenLiberty/open-liberty#14585]
I wanted to describe API response as follows:

```
@APIResponses({
            @APIResponse(responseCode = "200", content = {
                @Content(
                        mediaType = MediaType.TEXT_PLAIN,
                        schema = @Schema(implementation = MyEnum.class))})
    })
```
Api documentation is not generated at all.

I found out that the cause of the problem is enum as implementation inside @Schema.

If I replace `implementation = MyEnum.class` with `enumeration = {"ONE", "TWO"}`, everything works fine. But API should be generated dynamically.

Expected output:
```
responses:
        200:
          description: Success.
          content:
            text/plain:
              schema:
                enum:
                - ONE
                - TWO
```

**Diagnostic information:**
 - OpenLiberty Version: 20.0.0.10

**Comment by leochr** [comment]
@srnjak Does the referenced class `MyEnum.class` have the expected Schema annotations? Here is an example: https://github.com/eclipse/microprofile-open-api/blob/master/spec/src/main/asciidoc/microprofile-openapi-spec.adoc#schema

**Comment by srnjak** [comment]
@leochr It works same with or without `@Schema` annotation. Here is example with the annotation:

```
@Schema(name = "myEnum", description = "This is my enum.")
public enum MyEnum {
    ONE,
    TWO;
}
```

**Comment by leochr** [comment]
All values of a schema must specify a `type` and the type information is not available from the enum Java class. That could be the reason for this behaviour.

@msmiths do you have any thoughts on this?

**Comment by msmiths** [comment]
This issue can be resolved by moving to the `mpOpenAPI-2.0` feature. Using this feature and the annotations described in the issue, the OpenAPI document that is generated is as follows:

```
---
openapi: 3.0.3
info:
  title: Example REST API
  description: Example REST API
  version: "1.0"
servers:
- url: http://localhost:9080
- url: https://localhost:9443
paths:
  /actions/{actionId}:
    get:
      parameters:
      - name: actionId
        in: path
        description: The id of the action to retrieve.
        required: true
        schema:
          $ref: '#/components/schemas/UUID'
      responses:
        "200":
          description: Successful operation
          content:
            text/plain:
              schema:
                $ref: '#/components/schemas/MyEnum'
components:
  schemas:
    UUID:
      format: uuid
      pattern: "[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"
      type: string
    MyEnum:
      enum:
      - ONE
      - TWO
      type: string
```

**Comment by srnjak** [comment]
I can confirm that bug is resolved using `mpOpenAPI-2.0` using OpenLiberty `21.0.0.3`.
